utils/utils.jsx

```diff
@@ -2250,3 +2259,42 @@ export function deleteKeysFromObject(value, keys) {
     }
     return value;
 }
+
+export function isSelection() {
+    const selection = window.getSelection();
+    return selection.type === 'Range';
+}
+
+// Returns false when the element clicked or its ancestors
+// is a potential click target (link, button, image, etc..)
+// and true in any other case.
+export function isEligibleForClick(event) {
```

Comment:
0/5, but perhaps there's something that could be consolidated between this PR and @BenCookie95's Permalink Previews PR (#8304)? There's some similar logic [in this function](https://github.com/mattermost/mattermost-webapp/pull/8304/files#diff-13a27994b1a523f8b6c63e5085facd209a800c35c115d56469757cab7f55edb2R15) for determining whether or not you're clicking on a clickable element or somewhere else in the PostMessageView

Reply:
@hmhealey yes it's pretty similar, we could make this a function factory maybe so in addition to "clickable" elements we could pass additional class selectors depending on the case, what do you think?

*The only notable difference I see is that in this case we need to traverse parents of `event.target`s to ensure  nested elements (`span`s for example) in links or buttons are getting handled properly. If this is desired in the other PR as well we could generalize.

Reply:
I like this function, it is something I can re-use in my PR. The traversing is something that I need for the next part of the preview feature. Nice work!

Reply:
@BenCookie95, @hmhealey I've made some small changes, could you take a look?

I've changed `isEligibleForClick` to `makeIsEligibleForClick` which now takes a CSS selector as an argument and returns a `isEligibleForClick` function, so we are able to parametrize which other elements we don't want to handle the event for.
*CLICKABLE_ELEMENTS remain the same for all.